compiler/optimizer/Dominators.cpp

```diff
@@ -39,10 +36,11 @@
 #include "ras/Debug.hpp"                       // for TR_DebugBase
 
 TR_Dominators::TR_Dominators(TR::Compilation *c, bool post) :
+   _region(c->trMemory()->heapMemoryRegion()),
    _compilation(c),
-   _info(c->getFlowGraph()->getNextNodeNumber()+1, c->allocator(), c->allocator()),
-   _dfNumbers(c->getFlowGraph()->getNextNodeNumber()+1, 0, c->allocator()),
-   _dominators(c->getFlowGraph()->getNextNodeNumber()+1, static_cast<TR::Block *>(NULL), c->allocator())
+   _info(c->getFlowGraph()->getNextNodeNumber()+1, _region, _region),
```

Comment:
I think one of these `_region` parameters ends up being an implicit cast to a BBInfo.  It would be nice if this were made more obvious via an explicit construction-style cast.